SWIG/swap.i

```diff
@@ -364,4 +365,104 @@ class OvernightIndexedSwap : public Swap {
     const Leg& overnightLeg();
 };
 
+#if defined(SWIGPYTHON)
+%rename (_MakeOIS) MakeOIS;
+#endif
+%shared_ptr(MakeOIS)
+class MakeOIS {
+      public:
+        MakeOIS(const Period& swapTenor,
+                const boost::shared_ptr<OvernightIndex>& overnightIndex,
+                Rate fixedRate = Null<Rate>(),
+                const Period& fwdStart = 0*Days);
+
+        %extend{
+            boost::shared_ptr<OvernightIndexedSwap> makeOIS(){
+                return (boost::shared_ptr<OvernightIndexedSwap>)(* $self);
+            }
+        }
+
+        MakeOIS& receiveFixed(bool flag = true);
+        MakeOIS& withType(OvernightIndexedSwap::Type type);
+        MakeOIS& withNominal(Real n);
+        MakeOIS& withSettlementDays(Natural settlementDays);
+        MakeOIS& withEffectiveDate(const Date&);
+        MakeOIS& withTerminationDate(const Date&);
+        MakeOIS& withRule(DateGeneration::Rule r);
+        MakeOIS& withPaymentFrequency(Frequency f);
+        MakeOIS& withPaymentAdjustment(BusinessDayConvention convention);
+        MakeOIS& withPaymentLag(Natural lag);
+        MakeOIS& withPaymentCalendar(const Calendar& cal);
+        MakeOIS& withEndOfMonth(bool flag = true);
+        MakeOIS& withFixedLegDayCount(const DayCounter& dc);
+        MakeOIS& withOvernightLegSpread(Spread sp);
+        MakeOIS& withDiscountingTermStructure(
+                  const Handle<YieldTermStructure>& discountingTermStructure);
+        MakeOIS& withTelescopicValueDates(bool telescopicValueDates);
+        MakeOIS& withPricingEngine(
+                              const boost::shared_ptr<PricingEngine>& engine);
+};
+
+#if defined(SWIGPYTHON)
+%pythoncode{
+def MakeOIS(swapTenor, overnightIndex, fixedRate, fwdStart,
+            receiveFixed=None,
+            swapType=None,
+            nominal=None,
+            settlementDays=None,
+            effectiveDate=None,
+            terminationDate=None,
+            dateGenerationRule=None,
+            paymentFrequency=None,
+            paymentAdjustmentConvention=None,
+            paymentLag=None,
+            paymentCalendar=None,
+            endOfMonth=True,    
```

Comment:
Is that a good assumption? I guess it should be `True` for all swaps >= 1M, but `False` by default

Reply:
I would opt for consistency between C++ and Python, i.e. using the same default parameters.

Reply:
Code is updated and aligned with C++ default parameters. In contrary to MakeVanillaSwap I have added default parameters in Python signature of MakeOIS function (MakeVanillaSwap has None everywhere). By doing so, python developer will see in the function signature what are the default parameters hard coded within the MakeOIS constructor. The only exception that is missing, is setting `calendar_` to `overnightIndex->fixingCalendar()`. Exposing that would be quite tricky in swig (at least for me now) and would be probably of little value. Due to python interpreter, I also hide `fixedLegDayCount` setting behind None.